geode-apis-compatible-with-redis/src/test/java/org/apache/geode/redis/internal/data/RedisStringTest.java

```diff
@@ -351,4 +354,50 @@ public void setExpirationTimestamp_stores_delta_that_is_stable() throws IOExcept
     o2.fromDelta(in);
     assertThat(o2).isEqualTo(o1);
   }
+
+  @Test
+  public void overheadConstants_shouldNotChange_withoutForethoughtAndTesting() {
+    assertThat(RedisString.PER_OBJECT_OVERHEAD).isEqualTo(8);
+    assertThat(RedisString.getPerStringOverhead()).isEqualTo(RedisString.PER_STRING_OVERHEAD);
+  }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void should_calculateSize_closeToROSSize_ofIndividualInstances() {
+    String javaString;
+    for (int i = 0; i < 512; i += 8) {
+      javaString = makeStringOfSpecifiedSize(i);
+      RedisString string = new RedisString(new RedisKey(javaString.getBytes()));
+
+      int expected = reflectionObjectSizer.sizeof(string);
+      Long actual = Long.valueOf(string.getSizeInBytes());
+      Offset<Long> offset = Offset.offset(Math.round(expected * 0.05));
+
+      assertThat(actual).isCloseTo(expected, offset);
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void should_calculateSize_closeToROSSize_ofLargeStrings() {
+    String javaString = makeStringOfSpecifiedSize(10_000);
+
+    RedisKey byteArrayWrapper = new RedisKey(javaString.getBytes());
+    RedisString string = new RedisString(byteArrayWrapper);
+
+    Long actual = Long.valueOf(string.getSizeInBytes());
+    int expected = reflectionObjectSizer.sizeof(string);
+    Offset<Long> offset = Offset.offset(Math.round(expected * 0.05));
+
+    assertThat(actual).isCloseTo(expected, offset);
+  }
+
+  private String makeStringOfSpecifiedSize(final int stringSize) {
+    StringBuffer sb = new StringBuffer(stringSize);
+    for (int i = 0; i < stringSize; i++) {
+      sb.append("a");
+    }
+    String javaString = sb.toString();
+    return javaString;
+  }
```

Comment:
Do we need to add tests like the ones that are in `RedisHash` (e.g. sizeShouldChangeWhenValueIsChanged, shouldReturnToStringOverhead_whenValueIsRemoved, etc.)?